osu.Game/Overlays/BeatmapSet/BeatmapRulesetSelector.cs

```diff
@@ -0,0 +1,210 @@
+                Height = 1,
+            });
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colours)
+        {
+            bar.Colour = colours.Yellow;
+        }
+
+        /// <summary>
+        ///  Selects a <see cref="RulesetTabItem"/> that contains a matching <see cref="RulesetInfo"/>.
+        /// </summary>
+        public void SelectRuleset(RulesetInfo ruleset)
+        {
+            if (ruleset == null)
+                return;
+
+            foreach (var tab in TabContainer)
+                if (tab.Value.ID == ruleset.ID)
+                    SelectTab(tab);
+        }
+
+        private class RulesetTabItem : TabItem<RulesetInfo>
+        {
+            private readonly OsuSpriteText text;
+            private readonly Box hoverBox;
+
+            private readonly OsuSpriteText count;
+            private readonly Container countContainer;
+
+            private OsuColour colours;
+
+            public int BeatmapsCount
+            {
+                set
+                {
+                    Enabled.Value = value > 0;
+                    count.Text = Enabled.Value ? value.ToString() : null;
+
+                    updateState();
+                }
+            }
+
+            public override bool PropagatePositionalInputSubTree => Enabled.Value;
+
+            public RulesetTabItem(RulesetInfo value)
+                : base(value)
+            {
+                AutoSizeAxes = Axes.Both;
+
+                Children = new Drawable[]
+                {
+                    new FillFlowContainer
+                    {
+                        AutoSizeAxes = Axes.Both,
+                        Direction = FillDirection.Horizontal,
+                        Margin = new MarginPadding { Bottom = 10 },
+                        Children = new Drawable[]
+                        {
+                            text = new OsuSpriteText
+                            {
+                                Text = value.Name,
+                                Font = OsuFont.GetFont(),
+                            },
+                            countContainer = new Container
+                            {
+                                Margin = new MarginPadding { Left = 4 },
+                                AutoSizeAxes = Axes.Both,
+                                CornerRadius = 5,
+                                Masking = true,
+                                Children = new Drawable[]
+                                {
+                                    count = new OsuSpriteText
+                                    {
+                                        Font = OsuFont.GetFont(),
+                                        Padding = new MarginPadding { Left = 5, Right = 6, Bottom = 1 },
+                                    },
+                                },
+                            },
+                        },
+                    },
+                    hoverBox = new Box
```

Comment:
Why is this all not an `OsuTabControl`? This will require further refactoring - we can't be duplicating this logic all over the place.